docker/dockerProxyServer.js

```diff
@@ -0,0 +1,299 @@
+const version = 1.00;
+
+class DockerProxyServer {
+
+  constructor(port) {
+    this._serverPort = port || DEFAULT_SERVER_PORT;
+    this._requestId = 4000;
+    this._containerId = null;
+    this._server = null;
+    this._instanceLabel = 'dockerProxy' + (new Date()).toISOString();
+    this._originalNetworkPromise = this.getCurrentNetworks();
+  }
+
+  stopServer() {
+    if (this._server) {
+      this._server.close();
+      this._server = null;
+    }
+  }
+
+  startServer() {
+    return this.getCurrentNetworks().then((originalNetworks) => {
+      this._originalNetworks = originalNetworks;
+      return new Promise(resolve => {
+        const express = require('express');
+        const app = express();
+        app.use(cors());
+
+        [
+          { path: '/version', implfn: 'version' },
+          { path: '/isDocker', implfn: 'isDocker' },
+          { path: '/getCurrentContainerId', implfn: 'getCurrentContainerId' },
+          { path: '/getContainers', implfn: 'getContainers' },
+          { path: '/getActiveInterface', implfn: 'getActiveInterface' },
+          { path: '/disconnect/:networkId', implfn: 'disconnectFromNetwork' },
+          { path: '/disconnectFromAllNetworks', implfn: 'disconnectFromAllNetworks' },
+          { path: '/connect/:networkId', implfn: 'connectToNetwork' },
+          { path: '/createNetwork/:networkName', implfn: 'createNetwork' },
+          { path: '/resetNetwork', implfn: 'resetNetwork' },
+          { path: '/inspectCurrentContainer', implfn: 'inspectCurrentContainer' },
+          { path: '/connectToDefaultNetwork', implfn: 'connectToDefaultNetwork' },
+          { path: '/getAllNetworks', implfn: 'getAllNetworks' },
+          { path: '/getCurrentNetworks', implfn: 'getCurrentNetworks' },
+        ].forEach((route) => {
+          app.get(route.path, (req, res, next) => {
+            this[route.implfn](req.params).then(data => res.send(data)).catch(error => {
+              console.warn('Failed @ ' + route.path, error);
+              next(error);
+            });
+          });
+        });
+
+        this._server = app.listen(this._serverPort, () => {
+          console.log(`DockerProxyServer listening on port ${this._serverPort}!`);
+          resolve();
+        });
+      });
+    });
+  }
+
+  // resets network to default state
+  // default state is "twilio-video-default-network" as
+  // only connected network.
+  resetNetwork() {
+    return this.disconnectFromAllNetworks()
+        .then(() => this.connectToDefaultNetwork())
+        .then(() => this._pruneNetworks());
+  }
+
+  // removes all unused networks (created by this instance)
+  _pruneNetworks() {
+    const filters = encodeURIComponent(JSON.stringify({
+      'label': { [this._instanceLabel]: true }
+    }));
+    return this._makeRequest({
+      socketPath: '/var/run/docker.sock',
+      path: `/v1.32/networks/prune?filters=${filters}`,
+      method: 'POST',
+    });
+  }
+
+  getCurrentContainerId() {
```

Comment:
@makarandp0 docs?

Reply:
I understand docing client methods, but not sure if it's worth doc'ing these server internal methods. They are used internally by the webserver routes. I will mark them all as private for now.